### PR TITLE
Use native ruby plist serialization

### PR DESCRIPTION
**Pull request by segiddins**
Closes https://github.com/CocoaPods/CocoaPods/issues/5724

Removes dependencies on FFI, Xcode, and macOS.

Blocked by https://github.com/CocoaPods/Xcodeproj/pull/430

### CODE REVIEW

**Comment by segiddins** [comment]
At this point, I think the only failure is the change in display name for build phases!

**Comment by CocoaPodsBarista** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please include a CHANGELOG entry to credit yourself! 
You can find it at <a href="https://github.com/CocoaPods/CocoaPods/blob/master/CHANGELOG.md">CHANGELOG.md</a>.</td>
    </tr>
  </tbody>
</table>

Here's an example of your CHANGELOG entry:

```markdown
* Use native ruby plist serialization 
  [segiddins](https://github.com/segiddins)
  [#issue_number](https://github.com/CocoaPods/CocoaPods/issues/issue_number)
```

*note*: There are two invisible spaces after the entry's text.


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

**Comment by segiddins** [comment]
Hmm, I suppose we could only run danger on the lint build for now? Since I don't think the dangerfile is doing anything with the test results?

**Comment by segiddins** [comment]
Only legit failure is RealmSwift's `prepare_command` being reliant on `sh` actually invoking `bash`.
i.e. this is working. on linux. 3.5x faster than our old CI setup.

**Comment by DanToml** [comment]
😃💖

**Comment by orta** [comment]
> Hmm, I suppose we could only run danger on the lint build for now? 

Seems fine, can just make her run on one build - https://github.com/ruby-grape/grape/pull/1458

**Comment by segiddins** [comment]
`FileAccessor#dynamic_binary?` seems to be the only bottleneck at this point

**Comment by segiddins** [comment]
This is now 100% green. Using Nanaimo and running on Linux machines for CI (except for the examples).

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,7 +8,8 @@ To install release candidates run `[sudo] gem install cocoapods --pre`
 
 ##### Enhancements
 
-* None.  
+* Add support for running CocoaPods on Linux.
```
Comment (DanToml) [inline CHANGELOG.md]: Changelog message should include switching to native plist handling

Comment (DanToml) [inline CHANGELOG.md]: Also needs a link to the PR

File: lib/cocoapods/sandbox/file_accessor.rb [hunk 2]
```diff
@@ -1,3 +1,5 @@
+autoload :MachO, 'macho'
```
Comment (DanToml) [inline lib/cocoapods/sandbox/file_accessor.rb]: this should probably just be required in `dynamic_binary?`, as this is what we do for things like nap too.

File: .travis.yml [hunk 3]
```diff
@@ -1,35 +1,68 @@
...
+
 branches:
   only:
     - master
     - /.+-stable$/
 
 before_install:
+  - |
+      if [ "$COCOAPODS_CI_TASKS" != "LINT" ]; then
+        master_sha="$(ruby -e "puts '`git submodule status spec/fixtures/spec-repos/master`'.strip.sub(/^-/, '').split(/\s/, 2).first")"
```
Comment (DanToml) [inline .travis.yml]: this is 💯

**Comment by DanToml** [comment]
🎉

**Pull request merged**